resolver/resolver.cc

```diff
@@ -3708,11 +3721,15 @@ class ResolveSignaturesWalk {
         const auto &argv1 = args[1];
 
         // Unlike std::equal, absl::c_equal will test for equal sizes.
-        auto argsEqual = [&ctx](const auto &arg0, const auto &arg1) {
+        auto argsEqual = [&gs, owner](const auto &arg0, const auto &arg1) {
             if (arg0.name != arg1.name) {
                 return false;
             }
-            return core::Types::equiv(ctx, arg0.type, arg1.type);
+            auto arg0type =
+                core::Types::resultTypeAsSeenFrom(gs, arg0.type, owner, owner, owner.data(gs)->selfTypeArgs(gs));
```

Comment:
It might be worth experimenting with having `selfTypeArgs` return an inlined vector instead. I bet if we set it to `4` elements, we might avoid heap allocations most of the time.